### ISSUE TITLE
NPC Plumber Friend

### ISSUE THREAD

**Issue by uq-haze** [UQdeco2800/2022-studio-2#138]
# **Description**
**Task:** Creating a NPC Plumber Friend 
**Feature:** [NPC + Dialogue](https://github.com/UQdeco2800/2022-studio-2/issues/92)

A NPC which would be a friend of the player (lore-wise) where they will have a unique dialogue/interaction when the player interacts with them in the second map. The NPC will have a plunger and wrench that have been created by the weapons team (Team 4) to emphasise that it is a plumber. This ticket will contain drafts, iteration and animation of the NPC Plumber Friend.

# **Dependencies**
[Creation and Design of Level 1 Weapons ](https://github.com/UQdeco2800/2022-studio-2/issues/106)

- [x] Plunger
- [x] Wrench

# **Milestones**
List of steps that need to be completed for this task.
- [x] Step 1: Create Drafts (Sep 9)
- [x] Step 2: Create first iteration in pixel form (Sep 9)
- [x] Step 3: User testing (Sep 9)
- [x] Step 4: Second and Third Iterations (Sep 10)
- [x] Step 5: Create Animation and Documentation(Sep 11)
 
**Completion Deadline:** Sep 11

 
# **Documentation**
[NPC + Dialogue](https://github.com/UQdeco2800/2022-studio-2/wiki/Dialogue-&-NPCs)
 
# **Member**
- Hazeem bin Suffian @uq-haze 
- Caleb Ang @PurrplePandaa

**Comment by PurrplePandaa** [comment]
Hey @uq-haze , I've done some preliminary sketches for the wrench. What do you think?

![Spanner](https://user-images.githubusercontent.com/109809229/189510003-5ecd0817-75fd-483e-97fc-c4788244fcf1.jpg)
![wrench](https://user-images.githubusercontent.com/109809229/189510009-a9584fea-76e2-495b-af6f-64ea8af60a00.jpg)

**Comment by uq-haze** [comment]
Hey @PurrplePandaa, i like the drawings, however, i feel like the first image's metal part that is sticking out seems to resemble like a fish hook so I recommend removing it. Thanks!

**Comment by PurrplePandaa** [comment]
Hey yup, I was thinking that metal bit was supposed to be more of the bit where it sticks out cuz the spanner is all the way out. Idk if that makes sense but I've made the pixel drawing. First image on the left is what is was before, second image is what is was after the change you've suggested.

|![image](https://user-images.githubusercontent.com/109809229/189511086-b5c21bae-996f-4cf8-811d-c6ffb27b3f71.png)| 
![image](https://user-images.githubusercontent.com/109809229/189511088-35c4d881-5dea-4742-bfdd-fb0821c57232.png)|

Lmk how you feel about it @uq-haze ! 😄

**Comment by PurrplePandaa** [comment]
Hmm... mkay. What do you think about it now? I've kind of made the tip white so it makes a little more of a 3D effect whilst making the tips closer to each other. Though, I must admit, I'm not too sure how I feel about this one. Once again, lmk what your thoughts are in regards to the plumber friend holding this.

![image](https://user-images.githubusercontent.com/109809229/189511202-5c71f322-d450-45b3-b8ca-4bc18d4ae33a.png)

**Comment by uq-haze** [comment]
hmm...yea i kind of agree with you, maybe instead of the tips closer, we could try to make one tip longer than the other. Refer to bottom image as reference:
![Screenshot 2022-09-11 133643](https://user-images.githubusercontent.com/110069874/189511340-3a483e35-6c6d-4e25-b93f-6c0ef48c87ee.png)

**Comment by PurrplePandaa** [comment]
Ah yeap, okay gotcha.

**Comment by PurrplePandaa** [comment]
Alright, let's try this again. What do you think about this?
![image](https://user-images.githubusercontent.com/109809229/189511466-90e54cc9-8a7d-414d-addd-7112e83a24d9.png)

**Comment by uq-haze** [comment]
Aight, awesome bruv. If you could send me a pixel file, that would be great. Also maybe if you could add some animation to it that would look great in game

**Comment by PurrplePandaa** [comment]
Sweeeeeeeeet! Here ya go [Wrench.zip](https://github.com/UQdeco2800/2022-studio-2/files/9541731/Wrench.zip).
And also yeah, I was looking into the animations of this weapon and have also done some user testing to see how to go abouts doing that. However, this will seem like a sprint 3/4 item when detailing the game to look nicer in terms of weapons.

**Comment by uq-haze** [comment]
Aight cheers bruv!

**Comment by uq-haze** [comment]
Hey @PurrplePandaa, is it okay if I could ask for your pixel file for the plunger you made earlier in this sprint?

**Comment by PurrplePandaa** [comment]
Hey @uq-haze sure here is the file - [Plunger.zip](https://github.com/UQdeco2800/2022-studio-2/files/9541750/Plunger.zip). I also did some user testing regarding the plunger and had some people told me it might be good to differentiate the weapons between the player and NPCs/Enemies. I've already designed two different weapons coloured plungers in preparation for this but was thinking of implementing this within next sprint as our team has quite a lot of items on our plate atm.
